pkg/env/core.go

```diff
@@ -31,6 +34,20 @@ func (c *core) IsLocalDevelopmentMode() bool {
 	return c.isLocalDevelopmentMode
 }
 
+func (c *core) NewLiveConfigManager(ctx context.Context) (liveconfig.Manager, error) {
+	if c.isLocalDevelopmentMode {
+		return liveconfig.NewDev(), nil
+	} else {
+		msiAuthorizer, err := c.NewMSIAuthorizer(MSIContextRP, c.Environment().ResourceManagerEndpoint)
+		if err != nil {
+			return nil, err
+		}
+
+		mcc := containerservice.NewManagedClustersClient(c.Environment(), c.SubscriptionID(), msiAuthorizer)
```

Comment:
could we also read the kubeconfig once during deployment, so that the RP/GW MSI needs less runtime permissions?